binding/src/main/java/se/fortnox/reactivewizard/binding/scanners/InjectAnnotatedScanner.java

```diff
@@ -1,25 +1,27 @@
 package se.fortnox.reactivewizard.binding.scanners;
 
-import io.github.lukehutch.fastclasspathscanner.FastClasspathScanner;
+import io.github.classgraph.ClassInfo;
+import io.github.classgraph.ScanResult;
 
 import javax.inject.Singleton;
-import java.lang.reflect.Constructor;
-import java.lang.reflect.Executable;
 
 /**
  * Finds all classes with @Inject annotated constructors.
  */
 @Singleton
 public class InjectAnnotatedScanner extends AbstractClassScanner {
     @Override
-    public void visit(FastClasspathScanner fastClasspathScanner) {
-        fastClasspathScanner.matchClassesWithMethodAnnotation(javax.inject.Inject.class, this::classFound);
-        fastClasspathScanner.matchClassesWithMethodAnnotation(com.google.inject.Inject.class, this::classFound);
+    public void visit(ScanResult scanResult) {
```

Comment:
This will be a breaking change for every system that itself uses these scanners.
Should we try to make this change more flexible and hide the ScanResult behind an interface so that it will be easier to change the underlying tech in the future without disturbing implementing systems?

Reply:
That would be nice, but that interface would be designed only based on speculation on what future dependencies will provide. Also, it would require a wrapper for all classes used to represent the reflection (ClassInfo and MathodInfo and so on) and that interface would have to be adjusten as soon as someone would like to use any new feature of class graph. So I do think there is a poor ROI for such an abstraction.

Reply:
I am thinking a quite small change and no big investment.
A quick search shows we currently only use three different methods.
So the interface could be:
1. Iterable<Class<?>> findClassesAnnotatedWith(Class<?> annotationClass)
2. Iterable<Class<?>> findClassesWithMethodAnnotation(Class<?> annotationClass)
3. Iterable<Class<?>> findSubclassesOf(Class<?> class)
And the implementation of that could then use our classgraphscanner to provide these lists.

Reply:
fixed